projects/extension/src/components/Bootnodes.tsx

```diff
@@ -0,0 +1,262 @@
+  bootnode: string
+}
+
+interface Boot {
+  [key: string]: string[]
+}
+
+// Add to localstorage the given bootnode for the given chain
+const saveToLocalStorage = async (
+  chainName: string,
+  bootnode: string,
+  add: boolean,
+  def: string[],
+) => {
+  if (def.length === 0) throw new Error("Default Bootnodes should exist.")
+  let res: string[]
+  const chainBootnodes = await environment.get({
+    type: "bootnodes",
+    chainName,
+  })
+  res =
+    chainBootnodes && Object.keys(chainBootnodes).length > 0
+      ? [...chainBootnodes]
+      : [...def]
+  add ? res.push(bootnode) : res.splice(res.indexOf(bootnode), 1)
+  environment.set({ type: "bootnodes", chainName }, res)
+}
+
+export const Bootnodes = () => {
+  const [selectedChain, setSelectedChain] = useState<string>("polkadot")
+  const [defaultBn, setDefaultBn] = useState<BootnodesType[]>([])
+  const [customBn, setCustomBn] = useState<BootnodesType[]>([])
+  const [customBnInput, setCustomBnInput] = useState<string>("")
+  const [defaultWellKnownChainBn, setDefaultWellKnownChainBn] = useState<
+    string[]
+  >([])
+
+  const [addMessage, setAddMessage] = useState<any>(undefined)
+  const [bootnodeMsgClass, setBootnodeMsgClass] = useState<string>()
+
+  useEffect(() => {
+    // Load default Bootnodes
+    const defChains = environment.getDefaultBootnodes(selectedChain)
+    setDefaultWellKnownChainBn(defChains)
+  }, [selectedChain])
+
+  useEffect(() => {
+    if (addMessage && !addMessage?.error) {
+      setBootnodeMsgClass("pb-2 text-green-600")
+      setCustomBnInput("")
+    } else {
+      setBootnodeMsgClass("pb-2 text-red-600")
+    }
+  }, [addMessage])
+
+  useEffect(() => {
+    environment.getBootnodes(selectedChain).then((bootnodes) => {
+      const tmpDef: BootnodesType[] = []
+      const tmpCust: BootnodesType[] = []
+      // When bootnodes do not exist assign and save the local ones
+      if (!bootnodes?.length) {
+        environment.set(
+          { type: "bootnodes", chainName: selectedChain },
+          defaultWellKnownChainBn,
+        )
+        defaultWellKnownChainBn.forEach((b) => {
+          tmpDef.push({ bootnode: b, checked: true })
+        })
+      } else {
+        bootnodes?.forEach((b) => {
+          const defaultBootnodes =
+            environment.getDefaultBootnodes(selectedChain)
+          defaultBootnodes?.length && defaultBootnodes?.includes(b)
+            ? tmpDef.push({ bootnode: b, checked: true })
+            : tmpCust.push({ bootnode: b, checked: true })
+        })
+      }
+      setDefaultBn(tmpDef)
+      setCustomBn(tmpCust)
+    })
+  }, [selectedChain, defaultWellKnownChainBn])
+
+  const checkMultiAddr = (addr: string) => {
+    let regex58 =
+      /\/(ip4|ip6|dns4|dns6|dns)\/([a-zA-z0-9.-]{3,})\/tcp\/[0-9]{0,5}\/(ws|wss|tls|ws)\/p2p\/[a-zA-Z1-9^Il0O]{52}/i
+
+    const base64 =
+      /\/(ip4|ip6)\/([a-zA-z0-9.-]{3,})\/udp\/(.?)\/webrtc\/certhash\/(.*?)\/p2p\/[-A-Za-z0-9+=]{1,50}|=[^=]|={3,}/i
```

Comment:
Why is the thing after `/p2p/` different than for `regex58`? It's exactly the same format

Reply:
I was under the impression that the bootnode's structure of WebRTC is different. I think its better to alter those to match [the ones in smoldot](https://github.com/paritytech/smoldot/blob/main/bin/wasm-node/javascript/src/index-browser.ts#L88-L90)